### PR TITLE
Compat fixes

### LINKED ISSUE #19: Colors are not shown for outline items
For TypeScript, I do not get the colors:

image

### LINKED ISSUE #36: All I'm seeing is a bunch of undefined
image

I don't know why this is happening, but with Atom-TypeScript this is all I'm seeing on v1.3.1.

### PR DESCRIPTION
**Pull request by lierdakil**
Fixes #19, fixes #36

Do I need to manually commit build artefacts or are those built automatically?

### CODE REVIEW

**Comment by aminya** [comment]
Could you rebase with #35. I fixed the colors/icons there.

File: src/outlineView.js [hunk 1]
```diff
@@ -64,9 +64,9 @@ function addOutlineEntries({ parent, entries, editor, level = 0 }) {
     // Hold an entry in a dedicated element to prevent hover conflicts - hover over an <li> tag would be cought by a parent <li>
     const labelElement = document.createElement("span");
     labelElement.style.paddingLeft = `${10 * level}px`;
-    labelElement.innerText = item.representativeName;
+    labelElement.innerText = item.representativeName || item.plainText;
```
Comment (aminya) [inline src/outlineView.js:67]: This part is the missing part in #35

**Comment by aminya** [comment]
or I can cherry pick your commit.

**Comment by lierdakil** [comment]
I'm not keen on rebasing on #35. You've basically added a lot of questionable code for a one-line fix in #35 (see 5c330c2). Just why?

**Comment by aminya** [comment]
The main point is to allow user to supply `icon` as well as `kind`. The other part is to support the previous API of this package.

**Comment by lierdakil** [comment]
Yeah, I misunderstood how `icon` is supposed to work. Still, I think `icon` should supersede `kind`, so the code in #35 could be way easier I recon.

Anyway, just cherry-pick the commit.

**Comment by aminya** [comment]
I did. Check it out! 

By the way, I am interested in the folding feature of the semantics view of atom-typescript. If you can make a PR for that, it would be awesome.

**Comment by lierdakil** [comment]
atom-ide-ui actually handles collapsing nodes on its own:

atom-ide-ui's outline-view showing chevron arrows for collapsing nodes

I didn't do anything special there.

"Semantic view" feature in Atom-TypeScript uses a very different architecture, so porting parts of it won't be particularly easy. I'm afraid I won't be able to work on this anytime soon.

**Comment by aminya** [comment]
I should look into the code of both projects. Thanks for all of these clarifications! The thing that I was developing based on what I see in TypeScript's semantic view was shocking for me 😥😄

**Pull request closed without merging**